### PR TITLE
Ccc 12

### PR DESCRIPTION
**Pull request by triden92**

### CODE REVIEW

File: force-app/main/default/classes/JobApplicationHandler.cls [hunk 1]
```diff
@@ -85,4 +92,164 @@ public with sharing class JobApplicationHandler extends TriggerHandler {
...
+            if((socialSecurityCap != null && taxableIncome < socialSecurityCap) || socialSecurityCap == null){
+                socialSecurityWithheld = taxableIncome * socialSecurityRate;
+            }
+            else if(socialSecurityCap != null && taxableIncome >= socialSecurityCap){
+                socialSecurityWithheld = socialSecurityCap * socialSecurityRate;
+            }
+            if((medicareCap != null && taxableIncome < medicareCap) || medicareCap == null){
+                medicareWithheld = taxableIncome * medicareRate;
+            }
+            //Can uncomment lines 183-185 if needed. Accounts for cap on Medicare taxes, which currently don't exist
```
Comment (rileymeye1) [inline force-app/main/default/classes/JobApplicationHandler.cls]: @triden92 Remove the reference to line numbers, these could change in the future if someone updates the code. Keep the rest of your note about being able to uncomment. Maybe consider making it a comment block to group it so no one separates it if they were to update this code.

File: force-app/main/default/classes/JobApplicationHandlerTest.cls [hunk 2]
```diff
@@ -112,4 +112,225 @@ public with sharing class JobApplicationHandlerTest {
             }
         }
     }
+
+    @IsTest
+    static void testTaxWithholdingCalculator(){
+    
+        List<Job_Application__c> jobAppList = new List<Job_Application__c>();
+        Job_Application__c newJobApp1 = new Job_Application__c (Salary__c = 10000, Salary_Type__c = 'Annual', Tax_Filing_Status__c = 'Single', Deduction_Type__c = 'None');
```
Comment (rileymeye1) [inline force-app/main/default/classes/JobApplicationHandlerTest.cls:120]: @triden92 What if we add this to the TestDataFactory class? I'm already creating hundreds of records in mass with that.

File: force-app/main/default/classes/JobApplicationHandlerTest.cls [hunk 3]
```diff
@@ -112,4 +112,225 @@ public with sharing class JobApplicationHandlerTest {
...
+
+        }
+    }
+        
+
+
+    @IsTest
+    static void testdeductionUpdates(){
+        List<Job_Application__c> jobAppList = new List<Job_Application__c>();
+        Job_Application__c newJobApp1 = new Job_Application__c(Deduction_Type__c = 'Standard Deduction', Itemized_Deduction_Amount__c = 10000);
```
Comment (rileymeye1) [inline force-app/main/default/classes/JobApplicationHandlerTest.cls:283]: @triden92 same thing we could add to the TestDataFactory if you want to.

**Review by rileymeye1 — APPROVED** [review]
I left some comments all optional changes. Good to merge if you want to leave as is. @triden92

File: force-app/main/default/classes/JobApplicationHandlerTest.cls (hunk 2 above)
Comment (triden92) [reply force-app/main/default/classes/JobApplicationHandlerTest.cls:120]: @rileymeye1 I'm fine doing that. I'll take a look at the TestDataFactory and see how much work I need to do for that.

File: force-app/main/default/classes/JobApplicationHandlerTest.cls (hunk 3 above)
Comment (triden92) [reply force-app/main/default/classes/JobApplicationHandlerTest.cls:283]: @rileymeye1 I'm fine doing this one as well. Just need to take a look at the level of effort.

**Pull request merged**